### PR TITLE
Add resource deployment filtering by option values

### PR DESCRIPTION
**Pull request by Charles-Gagnon**
Previously the Arc dashboards had a feature where when you clicked a button on the dashboards to deploy a new instance it would open up the Resource Deployment wizard with only certain arc-related options shown.

But a recent change put MIAA under a new overall MI resource type and so we no longer can filter down to just Arc related items - it only allowed filtering to the resource type granularity.

This PR adds in additional filtering capabilities for the optionValues to only display certain values for those as well.

### CODE REVIEW

File: extensions/resource-deployment/src/ui/toolsAndEulaSettingsPage.ts [hunk 1]
```diff
@@ -216,6 +223,7 @@ export class ToolsAndEulaPage extends ResourceTypePage {
 						}));
 
 						this._optionDropDownMap.set(option.name, optionSelectBox);
+						this.wizard.provider = this.getCurrentProvider();
```
Comment (alanrenmsft) [inline extensions/resource-deployment/src/ui/toolsAndEulaSettingsPage.ts:226]: this line of change doesn't seem to be related to the option level filtering. Are you fixing another issue here?

Comment (Charles-Gagnon) [reply extensions/resource-deployment/src/ui/toolsAndEulaSettingsPage.ts:226]: Sort of. This is something that is called whenever you change the selected option to update the provider. Without it it just defaults to the first one - which wasn't necessarily correct when you filter out items. So it was never really an issue before because we only ever selected the first one anyways and then it would be updated correctly when another option was selected.

**Review by alanrenmsft — APPROVED** [review]
